### PR TITLE
feat(ras): mailchimp audience config

### PR DESCRIPTION
**Pull request by miguelpeixe**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Allow the Mailchimp audience to be configured for RAS integration. Also removes the `NEWSPACK_DATA_EVENTS_MAILCHIMP` feature flag.

image

### How to test the changes in this Pull Request:

1. Check out this branch and make sure you have Mailchimp as your ESP and RAS properly configured
2. Navigate to the "Newspack -> Engagement" page and while in the "Reader Activation" tab, click on "Show Advanced Settings"
3. Scroll down and confirm you see the Mailchimp integration section like in the image above
4. Select an audience and save
5. Refresh the page and confirm the saved audience persists
6. In a new session, register a new reader and make a donation
7. Wait until the event dispatches execute the handlers (with `define( 'NEWSPACK_LOG_LEVEL', 2 );` you should see:
```
[NEWSPACK-DATA-EVENTS][Newspack\Data_Events::handle]: Executing global action handlers for "donation_new".
```
8. Go to your Mailchimp audience and confirm the contact is updated with `NP_` prefixed metadata

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: assets/wizards/engagement/components/mailchimp.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import { __ } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { Notice, SectionHeader, SelectControl } from '../../../components/src';
+
+export default function ActiveCampaign( { value, onChange } ) {
```
Comment (dkoo) [inline assets/wizards/engagement/components/mailchimp.js:13]: paste error?
```suggestion
export default function Mailchimp( { value, onChange } ) {
```

**Review by dkoo — CHANGES_REQUESTED** [review]
Just a minor typo, otherwise it works great! Another thing to note: this is probably expected because Mailchimp doesn't have the concept of a "master list" like AC (which contacts are added to but don't receive newsletter content from). If you donate in a new session without registering or signing up for a newsletter first, the contact will not be synced to Mailchimp. If you subsequently sign up for a newsletter, your donation data will be synced after the fact.

**Comment by miguelpeixe** [comment]
Thanks, Derrick!

> If you donate in a new session without registering or signing up for a newsletter first, the contact will not be synced to Mailchimp.

That's not the expected behavior. Mailchimp supports transactional contacts, which is how we handle this API request, with an upsert strategy with `put()` and `status_if_new` set to `transactional`:

https://github.com/Automattic/newspack-plugin/blob/412a241562a70c40bb070858124095325fb669df/includes/data-events/connectors/class-mailchimp.php#L131-L158

Looking into it.

**Comment by miguelpeixe** [comment]
I'm not sure if that was the case for you, but I caught an error with my local instance while generating the merge fields. My MC account has reached the limit of merge fields and creating a new field was returning an API error.

4596d7db20f3f29f39d266354fc6cade05110a31 skips fields that failed to create in order to allow the data push

**Review by dkoo — CHANGES_REQUESTED** [review]
It seems to be working now, thanks! I didn't see any error before, but I'm seeing non-registered donors being synced immediately now.

I did notice some bugginess in the data syncing, though:

- The `NP_Total Paid` seems to be double what I'd expect after making a single one-time or recurring donation. e.g. if I donate $90 one time, this gets synced as $180. If I sign up for a $15 monthly subscription, it gets synced as $30.
- If I make subsequent donations using the same account, the `NP_Total Paid` amount updates incorrectly as well. With the recurring test donor, I made a $15 monthly donations, and `NP_Total Paid` got synced at first as $30. I then made a $9 one-time donation, and it got synced to $33.
- Based on this behavior I think we're adding the last payment amount to the total spend, since I would have expected the total to be $24 after the second donation.

**Comment by miguelpeixe** [comment]
Can you confirm that this is only happening in the Mailchimp connector? This looks like a problem in the logic that generates the data, which is a separate code that also feeds ActiveCampaign's integration.

Trying to reproduce now

**Comment by miguelpeixe** [comment]
Yep, reproduced with a new account donating $90 one-time, results in a $180 as "total paid". After paying an additional $20, the "total paid" changes to $130 😕

**Review by dkoo — APPROVED** [review]
I see similar but slightly different behavior with ActiveCampaign. The total after the first donation is doubled, just like in Mailchimp. After subsequent donations, the total comes in correctly. But since this seems unrelated to Mailchimp specifically, let's handle that in a separate hotfix PR. This one looks good!

**Pull request merged**